fuzzers/libfuzzer_libpng_centralized/src/lib.rs

```diff
@@ -135,6 +135,114 @@ pub extern "C" fn libafl_main() {
     let shmem_provider = StdShMemProvider::new().expect("Failed to init shared memory");
 
     let monitor = MultiMonitor::new(|s| println!("{s}"));
+    let mut main_run_client = |state: Option<_>, mut mgr, _core_id: CoreId| {
```

Comment:
So having two closures means we need to copy&paste a bunch of stuff? That's also not perfect..
Why not use a flag in the `mgr`?